src/Tribe/Venue.php

```diff
@@ -0,0 +1,368 @@
+					$data['StateProvince'] = $data['Province'];
+				} else {
+					$data['StateProvince'] = '';
+				}
+			}
+		}
+
+		update_post_meta( $venue_id, '_EventShowMapLink', isset( $data['EventShowMapLink'] ) );
+		update_post_meta( $venue_id, '_EventShowMap', isset( $data['EventShowMap'] ) );
+		unset( $data['EventShowMapLink'] );
+		unset( $data['EventShowMap'] );
+
+		if ( isset( $data['FeaturedImage'] ) && ! empty( $data['FeaturedImage'] ) ) {
+			update_post_meta( $venue_id, '_thumbnail_id', $data['FeaturedImage'] );
+			unset( $data['FeaturedImage'] );
+		}
+
+		foreach ( $data as $key => $var ) {
+			update_post_meta( $venue_id, '_Venue' . $key, $var );
+		}
+	}
+
+	/**
+	 * Creates a new venue
+	 *
+	 * @param array  $data        The venue data.
+	 * @param string $post_status the intended post status.
+	 *
+	 * @return mixed
+	 */
+	public function create( $data, $post_status = 'publish' ) {
+
+		if ( ( isset( $data['Venue'] ) && $data['Venue'] ) || $this->has_venue_data( $data ) ) {
+			$title   = isset( $data['Venue'] ) ? $data['Venue'] : esc_html__( 'Unnamed Venue', 'the-events-calendar' );
+			$content = isset( $data['Description'] ) ? $data['Description'] : '';
+			$slug    = sanitize_title( $title );
+
+			$postdata = array(
+				'post_title'  => $title,
+				'post_content' => $content,
+				'post_name'   => $slug,
+				'post_type'   => self::POSTTYPE,
+				'post_status' => $post_status,
+			);
+
+			$venue_id = wp_insert_post( $postdata, true );
+
+			// By default, the show map and show map link options should be on
+			$data['ShowMap'] = isset( $data['ShowMap'] ) ? $data['ShowMap'] : 'true';
+			$data['ShowMapLink'] = isset( $data['ShowMapLink'] ) ? $data['ShowMapLink'] : 'true';
+
+			if ( ! is_wp_error( $venue_id ) ) {
+				$this->save_meta( $venue_id, $data );
+				do_action( 'tribe_events_venue_created', $venue_id, $data );
+
+				return $venue_id;
+			}
+		}
+
+		// if the venue is blank, let's save the value as 0 instead
+		return 0;
+	}
+
+	/**
+	 * Check to see if any venue data set
+	 *
+	 * @param array $data the venue data.
+	 *
+	 * @return bool If there is ANY venue data set, return true.
+	 */
+	public function has_venue_data( $data ) {
+		foreach ( self::$valid_venue_keys as $key ) {
+			if ( isset( $data[ $key ] ) && $data[ $key ] ) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Updates an venue
+	 *
+	 * @param int   $venue_id The venue ID to update.
+	 * @param array $data    The venue data.
+	 *
+	 */
+	public function update( $venue_id, $data ) {
+		$data['ShowMap']     = isset( $data['ShowMap'] ) ? $data['ShowMap'] : 'false';
+		$data['ShowMapLink'] = isset( $data['ShowMapLink'] ) ? $data['ShowMapLink'] : 'false';
+
+		$this->save_meta( $venue_id, $data );
+		do_action( 'tribe_events_venue_updated', $venue_id, $data );
+	}
+
+	/**
+	 * Deletes a venue
+	 *
+	 * @param int  $venue_id      The venue ID to delete.
+	 * @param bool $force_delete Same as WP param.
```

Comment:
By default isn't `$force_delete` largely useless for CPTs? The comment notes it works as per the WP param but I wonder if we should give a clearer hint (especially since it's a public method).